Update subreddit.py

xCROv opened the pull request:
Documenting required parameter `flair_id` when using `flair_text`. When looking at this we may need to add this to all submission functions but I have not tested this as of yet so will submit another PR later if that is the case.

Fixes #66

## Feature Summary and Justification

Documenting required atributes

## Notes

This will probably need to be done with all submit functions but I have not tested them yet to be able to make that PR.

LilSpazJoekp commented on asyncpraw/models/reddit/subreddit.py:
```diff
@@ -913,7 +913,7 @@ async def submit(
         :param flair_id: The flair template to select (default: None).
         :param flair_text: If the template's ``flair_text_editable`` value is
-            True, this value will set a custom text (default: None).
+            True, this value will set a custom text (default: None) (``flair_id`` required).
```
```suggestion
            True, this value will set a custom text (default: None). ``flair_id`` is
            required when ``flair_text`` is provided``
```

LilSpazJoekp commented:
Just a minor tweak.

LilSpazJoekp commented:
Could you add yourself to the documentation contributors section in AUTHORS.rst?

xCROv commented:
> Could you add yourself to the documentation contributors section in AUTHORS.rst?

Added. I have also updated the other submit functions to reflect your tweak to the first one.

LilSpazJoekp commented:
Could you squash your commits to a single commit?

LilSpazJoekp commented:
What IDE do you use?

xCROv commented:
Never squashed comments on the PR side and have probably made it more complicated now. 🤣

LilSpazJoekp commented on asyncpraw/models/reddit/subreddit.py:
```diff
@@ -913,7 +913,8 @@ async def submit(
         :param flair_text: If the template's ``flair_text_editable`` value is
-            True, this value will set a custom text (default: None).
+            True, this value will set a custom text (default: None). ``flair_id`` is
+            required when ``flair_text`` is provided``
```
```suggestion
            required when ``flair_text`` is provided.
```

LilSpazJoekp commented on asyncpraw/models/reddit/subreddit.py:
```diff
@@ -1047,7 +1048,8 @@ async def submit_gallery(
             traditional comments (default: None).
         :param flair_id: The flair template to select (default: None).
         :param flair_text: If the template's ``flair_text_editable`` value isTrue, this
-            value will set a custom text (default: None).
+            value will set a custom text (default: None). ``flair_id`` is
+            required when ``flair_text`` is provided``
```
```suggestion
        :param flair_text: If the template's ``flair_text_editable`` value is
              True, this value will set a custom text (default: None). ``flair_id`` is
              required when ``flair_text`` is provided.
```

LilSpazJoekp commented on asyncpraw/models/reddit/subreddit.py:
```diff
@@ -1157,7 +1159,8 @@ async def submit_image(
         :param flair_text: If the template's ``flair_text_editable`` value is
-            True, this value will set a custom text (default: None).
+            True, this value will set a custom text (default: None). ``flair_id`` is
+            required when ``flair_text`` is provided``
```
```suggestion
            required when ``flair_text`` is provided.
```

LilSpazJoekp commented on asyncpraw/models/reddit/subreddit.py:
```diff
@@ -1267,7 +1270,8 @@ async def submit_poll(
         :param flair_text: If the template's ``flair_text_editable`` value is
-            True, this value will set a custom text (default: None).
+            True, this value will set a custom text (default: None). ``flair_id`` is
+            required when ``flair_text`` is provided``
```
```suggestion
            required when ``flair_text`` is provided.
```

LilSpazJoekp commented on asyncpraw/models/reddit/subreddit.py:
```diff
@@ -1344,7 +1348,8 @@ async def submit_video(
         :param flair_text: If the template's ``flair_text_editable`` value is
-            True, this value will set a custom text (default: ``None``).
+            True, this value will set a custom text (default: ``None``). ``flair_id`` is
+            required when ``flair_text`` is provided``
```
```suggestion
            required when ``flair_text`` is provided.
```
